src/basic/process-util.c

```diff
@@ -169,7 +169,7 @@ int get_process_cmdline(pid_t pid, size_t max_columns, ProcessCmdlineFlags flags
                         if (t[i] == '\0')
                                 t[i] = ' ';
 
-                t[k] = '\0'; /* Normally, t[k] is already NUL, so this is just a guard in case of short read */
+                t[k - 1] = '\0'; /* Normally, t[k - 1] is already NUL, so this is just a guard in case of short read */
```

Comment:
Hmm, wasn't this already fixed by 0c13b949c3?

Reply:
Oh, OK, it's part of #18863 which hasn't been merged yet.
I think my fix is nicer. Of course this doesn't matter much, but conceptually it's nicer to read the precise number of bytes, instead of reading one extra and then discarding it.

But we need to resolve this before v248, and #18863 will not go in yet.
So maybe you could replace the first commit in this PR with the matching one from #18863?

Reply:
Actually, read_full_virtual_file() does NUL termination on it's own already. So this line here should be dropped.